src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -197,4 +211,82 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    private enum State {
+        ADDRESS_BOOK,
+        PROJECT_LIST
+    }
+
+    private void changeUiDisplay(State nextState) {
+        switch (nextState) {
+        case ADDRESS_BOOK:
+            personListPanel = new PersonListPanel(logic.getFilteredPersonList());
+            projectListPanelPlaceholder.getChildren().setAll(personListPanel.getRoot());
+            currentState = nextState;
+            break;
+
+        case PROJECT_LIST:
+            projectListPanel = new ProjectListPanel(logic.getFilteredProjectList());
+            projectListPanelPlaceholder.getChildren().add(projectListPanel.getRoot());
+            currentState = nextState;
+            break;
+
+        default:
+            assert false : "Unrecognised state";
+        }
+    }
+
+    private State stateOf(String commandWord) {
```

Comment:
Maybe can let each command class to have a attribute state so this class not so bulky